src/metabase/api/task.clj

```diff
@@ -21,7 +21,8 @@
 (api/defendpoint GET "/:id"
   "Get `TaskHistory` entry with ID."
   [id]
-  (api/read-check TaskHistory id))
+  (validation/check-has-general-permission :monitoring)
+  (api/check-404 (TaskHistory id)))
```

Comment:
For `TaskHistory` it might be good to just change the permissions at the model level, since it seems like users with monitoring permissions should always have read perms for this model.

i.e. you can update the model's implementation of `i/IObjectPermissions` so that read checks happen against a custom `perms-objects-set` which is either `#{"/general/monitoring/"}` if advanced perms are enabled, or just `#{"/"}` otherwise. As an example, look at how Table perms are implemented in `metabase.models.table`.

Then you can just call `api/read-check` and it'll automatically enforce the appropriate permissions.

Reply:
good point, I've added it